scripts/dev/estimate-usage/estimate-usage.armv7-a.safemode.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -e
+
+print_fs_usage() {
+    local input_image="$1"
+    local image="$input_image"
+    if test ! -f "$image"; then
+	# We need to look at its filesize and read it multiple times, so if it's
```

Comment:
I don't follow this. Can you explain?

Reply:
These scripts can be used as part of a pipeline (in fact the top-level script uses them that way to save some disk I/O for the VM). Unlike the others, this subscript involves reading the file size (with `du --apparent-size` a couple lines later) and its content twice (with `dumpimage`). If the image given is not a plain file (e.g. a symbolic link or FIFO) the script will fail (producing an incorrect answer for the file size, and for FIFOs it will fail with an invalid format error for the second invocation of `dumpimage`, as the first invocation consumed the input already).  
It's probably possible to use `tee` to avoid this, but I figured that'd be more difficult to read.

The other scripts are unaffected by this consideration as they only read the image once, when unpacking the tarball.